pkg/disruptors/pod.go

```diff
@@ -104,13 +108,61 @@ func (d *podDisruptor) Targets() ([]string, error) {
 func (d *podDisruptor) InjectHTTPFaults(fault HTTPFault, duration uint, options HTTPDisruptionOptions) error {
 	cmd := buildHTTPFaultCmd(fault, duration, options)
 
-	err := d.controller.ExecCommand(cmd)
+	err := d.validateTargetPort(fault.Port)
+	if err != nil {
+		return err
+	}
+
+	err = d.controller.ExecCommand(cmd)
 	return err
 }
 
 // InjectGrpcFaults injects faults in the grpc requests sent to the disruptor's targets
 func (d *podDisruptor) InjectGrpcFaults(fault GrpcFault, duration uint, options GrpcDisruptionOptions) error {
 	cmd := buildGrpcFaultCmd(fault, duration, options)
-	err := d.controller.ExecCommand(cmd)
+
+	err := d.validateTargetPort(fault.Port)
+	if err != nil {
+		return err
+	}
+
+	err = d.controller.ExecCommand(cmd)
 	return err
 }
+
+// validateTargetPort checks if every target pod listens to target port
+func (d *podDisruptor) validateTargetPort(targetPort uint) error {
+	if targetPort == 0 {
+		targetPort = protocol.DefaultTargetPort
```

Comment:
I agree that having two defaults is like a parallel API. I didn't understand the idea behind default 80 on the agent side and I also think that it would be better to make it a required parameter 👍  
I will create a follow-up issue